skrub/_table_vectorizer.py

```diff
@@ -1125,28 +958,57 @@ def get_feature_names_out(self, input_features=None) -> list[str]:
         list of str
             Feature names.
         """
-        ct_feature_names = super().get_feature_names_out()
-        all_trans_feature_names = []
+        return self._column_transformer.get_feature_names_out(input_features)
 
-        for name, trans, cols, _ in self._iter(
-            fitted=True, replace_strings=False, column_as_strings=False
-        ):
-            if isinstance(trans, str):
-                if trans == "drop":
-                    continue
-                elif trans == "passthrough":
-                    if all(isinstance(col, int) for col in cols):
-                        cols = [self.columns_[i] for i in cols]
-                    all_trans_feature_names.extend(cols)
-                continue
-            trans_feature_names = trans.get_feature_names_out(cols)
-            all_trans_feature_names.extend(trans_feature_names)
-
-        if len(ct_feature_names) != len(all_trans_feature_names):
-            warn("Could not extract clean feature names; returning defaults. ")
-            return list(ct_feature_names)
-
-        return all_trans_feature_names
+    @property
+    def transformers_(self):
+        """Transformers applied to the different columns."""
+        # For the "remainder" columns, the `ColumnTransformer` `transformers_`
+        # attribute contains the index instead of the column name,
+        # so we convert the values to the appropriate column names
+        # if there is less than 20 columns in the remainder.
+        transformers = self._column_transformer.transformers_
+        for i, (name, transformer, columns) in enumerate(transformers):
+            if name == "remainder" and len(columns) < 20:
+                # In this case, "columns" is a list of ints (the indices)
+                columns: list[int]
+                transformers[i] = (
```

Comment:
```python
import pandas as pd
from skrub import TableVectorizer

X = pd.DataFrame({"a": [True, False]})

vectorizer = TableVectorizer().fit(X)
print(vectorizer.transformers_)
print(vectorizer.transformers_) # IndexError

```

Reply:
Yep we would need to copy here.

Reply:
btw IIUC "remainder" can only apply to booleans? maybe we should mention that in the doc

Reply:
also I would make it always column names -- I find it very surprising and I think it can lead to bugs if one of the entries in this list has a different meaning than the rest, and also that this can change by adding or removing 1 column to the input

Reply:
thanks for opening https://github.com/scikit-learn/scikit-learn/issues/27533 @glemaitre

Reply:
> btw IIUC "remainder" can only apply to booleans? maybe we should mention that in the doc

what do you mean?

Reply:
just that as numeric and categorical are treated separately, few types are left to be handled directly in the "remainder" section (of course we can redirect other types by passing eg `numeric="remainder"`. the only one I found while trying to trigger the error mentioned [here](https://github.com/skrub-data/skrub/pull/761#discussion_r1346119765) was boolean. It might be useful to tell a user who chooses a transformer for the "remainder" argument what types of columns will fall into this category. (it is different from the ColumnVectorizer for which "remainder" are all the columns we haven't listed explicitly)

Reply:
I see, thanks for clarifying. Should we set boolean columns to `passthrough` like we do by default for numerical columns? It feels weird to me to isolate a specific type in `remainder`.

Reply:
yes I think it's maybe better to treat boolean together with the numerical columns, and in this case we probably don't need the "remainder" anymore

Reply:
continued in https://github.com/skrub-data/skrub/discussions/793